### PR TITLE
WIP: update docstring

### PR DESCRIPTION
**Pull request by chiahaoliu**
This is a PR in progress. It's for adding docstring to public and used API to `xpdacq`.

Please feel free to correct my wording and typo since I know there will be huge room for it :p

### CODE REVIEW

File: xpdacq/beamtime.py [hunk 1]
```diff
@@ -77,6 +77,25 @@ def _configure_pe1c(exposure):
 
 
 def ct(dets, exposure, *, md=None):
+    """
+    Take one reading from area detectors with given exposure time
+
+    Parameters
+    ----------
+    detectors : list
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: shouldn't this  be `dets`?

File: xpdacq/beamtime.py [hunk 2]
```diff
@@ -77,6 +77,25 @@ def _configure_pe1c(exposure):
...
+        list of 'readable' objects
+    exposure : float
+        total time of exposrue in seconds
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: typo

File: xpdacq/beamtime.py [hunk 3]
```diff
@@ -77,6 +77,25 @@ def _configure_pe1c(exposure):
 
 
 def ct(dets, exposure, *, md=None):
+    """
+    Take one reading from area detectors with given exposure time
+
+    Parameters
+    ----------
+    detectors : list
+        list of 'readable' objects
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: more info?  I guess I infer that it is bluesky detector objects, or what?

File: xpdacq/beamtime.py [hunk 4]
```diff
@@ -77,6 +77,25 @@ def _configure_pe1c(exposure):
...
+    exposure : float
+        total time of exposrue in seconds
+    md : dict, optional
+        extra metadata
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: extra metadata what?  add more. Basically, if md is specified here, when will it be saved?  Every time ct is run, or just this instance?

File: xpdacq/beamtime.py [hunk 5]
```diff
@@ -77,6 +77,25 @@ def _configure_pe1c(exposure):
...
+
+    Note
+    ----
+    area detector that is triggered will always be the one configured in
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: how does this relate to `dets` that is passed as an argument though?  It makes it confusing what is happening

File: xpdacq/beamtime.py [hunk 6]
```diff
@@ -98,6 +117,33 @@ def ct(dets, exposure, *, md=None):
 
 
 def Tramp(dets, exposure, Tstart, Tstop, Tstep, *, md=None):
+    """
+    Scan over temeprature controller in steps.
+
+    temeprature steps are defined by starting point, stoping point and step size
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: typo

File: xpdacq/beamtime.py [hunk 7]
```diff
@@ -98,6 +117,33 @@ def ct(dets, exposure, *, md=None):
 
 
 def Tramp(dets, exposure, Tstart, Tstop, Tstep, *, md=None):
+    """
+    Scan over temeprature controller in steps.
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: typo

File: xpdacq/beamtime.py [hunk 8]
```diff
@@ -98,6 +117,33 @@ def ct(dets, exposure, *, md=None):
...
+
+    temeprature steps are defined by starting point, stoping point and step size
+
+    Parameters
+    ----------
+    detectors : list
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: dets?

File: xpdacq/beamtime.py [hunk 9]
```diff
@@ -98,6 +117,33 @@ def ct(dets, exposure, *, md=None):
...
+        list of 'readable' objects
+    exposure : float
+        exposure time at each temeprature step in seconds
+    Tstart : float
+        starting point of temperature sequence
+    Tstop : float
+        stoping point of temperature sequence
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: typo

File: xpdacq/beamtime.py [hunk 10]
```diff
@@ -98,6 +117,33 @@ def ct(dets, exposure, *, md=None):
...
+    Scan over temeprature controller in steps.
+
+    temeprature steps are defined by starting point, stoping point and step size
+
+    Parameters
+    ----------
+    detectors : list
+        list of 'readable' objects
+    exposure : float
+        exposure time at each temeprature step in seconds
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: typo

File: xpdacq/beamtime.py [hunk 11]
```diff
@@ -98,6 +117,33 @@ def ct(dets, exposure, *, md=None):
...
+    Tstart : float
+        starting point of temperature sequence
+    Tstop : float
+        stoping point of temperature sequence
+    Tstep : float
+        step size between Tstart and Tstop of this sequence
+    md : dict, optional
+        extra metadata
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: see comment above

File: xpdacq/beamtime.py [hunk 12]
```diff
@@ -98,6 +117,33 @@ def ct(dets, exposure, *, md=None):
...
+
+    Note
+    ----
+    temeprature controller that is driven will always be the one configured in
```
Comment (sbillinge) [inline xpdacq/beamtime.py]: typo

File: xpdacq/beamtime.py [hunk 13]
```diff
@@ -183,6 +250,42 @@ def _clean_info(obj):
 
 
 class Beamtime(ValidatedDictLike, YamlDict):
+    """
+    class that carries necessary information for a beamtime
+
+    Parameters
+    ----------
+    pi_last : str
+        last name of PI to this beamtime.
```
Comment (CJ-Wright) [inline xpdacq/beamtime.py]: PI for this beamtime?

File: xpdacq/beamtime.py [hunk 14]
```diff
@@ -183,6 +250,42 @@ def _clean_info(obj):
...
+    saf_num : int
+        Safty Approval Form number to current beamtime.
```
Comment (CJ-Wright) [inline xpdacq/beamtime.py]: for current beamtime

**Comment by chiahaoliu** [comment]
close as we will do this in separate Pos

**Pull request closed without merging**